packages/cojson/src/sync.ts

```diff
@@ -551,6 +551,14 @@ export class SyncManager {
 
     let invalidStateAssumed = false;
 
+    const contentToStore: NewContentMessage = {
+      action: "content",
+      id: msg.id,
+      priority: msg.priority,
+      header: msg.header,
+      new: {},
+    };
```

Comment:
Moved the content filtering in `handleNewContent` because:
- This is actually the only place were we want to do this, `storeContent` is also called when we do store new local transactions, doing the filtering there means unnecessary computation
- After reviewing the PR I realized that we were also storing values with invalid transactions, which means we need filter out invalid sessions not only by knownState check, but also by `tryAddTransaction` success

Since this has just become an higher priority issue, going to merge this right away after adding some more tests.